### PR TITLE
Fixed getPackageInfo deprecation.

### PR DESCRIPTION
**Pull request by n1snt**
## Purpose / Description
Fixes `getPackageInfo` depreciation issue in https://github.com/ankidroid/Anki-Android/issues/12364#issuecomment-1264426702

## How Has This Been Tested?

- Tests
- Physical device
- API 33 Emulator

## Checklist

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/utils/VersionUtils.kt [hunk 1]
```diff
@@ -37,7 +38,11 @@ object VersionUtils {
             var pkgName = AnkiDroidApp.TAG
             val context: Context = applicationInstance ?: return AnkiDroidApp.TAG
             try {
-                val pInfo = context.packageManager.getPackageInfo(context.packageName, 0)
+                val pInfo = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/utils/VersionUtils.kt:41]: ditto of my other review, this needs to be in `compat`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/utils/VersionUtils.kt (hunk 1 above)
Comment (n1snt) [reply AnkiDroid/src/main/java/com/ichi2/utils/VersionUtils.kt:41]: Thanks, will move this to `compat` once I get home.

Comment (n1snt) [reply AnkiDroid/src/main/java/com/ichi2/utils/VersionUtils.kt]: Implemented it using Compat can you please have a look at it?

**Comment by n1snt** [comment]
@david-allison made the requested changes.

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt [hunk 2]
```diff
@@ -83,5 +85,9 @@ class CompatHelper private constructor() {
         inline fun <reified T : Parcelable> Intent.getParcelableExtraCompat(name: String): T? {
             return compat.getParcelableExtra(this, name, T::class.java)
         }
+
+        fun getPackageInfoCompat(context: Context, packageName: String, flags: Int): PackageInfo {
+            return compat.getPackageInfoCompat(context, packageName, flags)
+        }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt:91]: The point of these functions is that they're extensions. This should likely be an extension on `Context`

File: AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt [hunk 3]
```diff
@@ -92,6 +94,9 @@ interface Compat {
      */
     fun <T : Parcelable?> getParcelableExtra(intent: Intent, name: String, clazz: Class<T>): T?
 
+    @Throws(NameNotFoundException::class)
+    fun getPackageInfoCompat(context: Context, packageName: String, flags: Int): PackageInfo
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt:98]: Should this be `@PackageInfoFlagsBits flags`

File: AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt [hunk 4]
```diff
@@ -92,6 +94,9 @@ interface Compat {
      */
     fun <T : Parcelable?> getParcelableExtra(intent: Intent, name: String, clazz: Class<T>): T?
 
+    @Throws(NameNotFoundException::class)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt:97]:
```suggestion
    /**
     * Retrieve overall information about an application package that is
     * installed on the system.
     * 
     * @see PackageManager.getPackageInfo
     */
    @Throws(NameNotFoundException::class)
```

**Review by david-allison — CHANGES_REQUESTED** [review]
Much better! Some code style improvements/docs remaining

**Review by david-allison — CHANGES_REQUESTED** [review]
This doesn't seem to be all of the usages: 

Screenshot 2022-10-25 at 11 15 47

File: AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt (hunk 4 above)
Comment (n1snt) [reply AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt]: Done.

File: AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt (hunk 2 above)
Comment (n1snt) [reply AnkiDroid/src/main/java/com/ichi2/compat/CompatHelper.kt]: Done.

File: AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt (hunk 3 above)
Comment (n1snt) [reply AnkiDroid/src/main/java/com/ichi2/compat/Compat.kt:104]: Should I change 
`@Throws(NameNotFoundException::class)`
with 
`@PackageInfoFlagsBits flags`? 🫤

**Comment by n1snt** [comment]
@david-allison can you please have a look at the changes I made?
There is one usage left in isSystemApp in AdaptationUtil & I'm not sure how to use compact there as it's accessing getPackageInfo from the PackageManager which is passed to it as an argument.

**Comment by n1snt** [comment]
Hey @david-allison can I please get a review on this one?

**Pull request closed without merging**